pallets/asset-manager/src/tests.rs

```diff
@@ -189,3 +165,50 @@ fn update_asset() {
 		);
 	})
 }
+
+#[test]
+fn filter_asset_location_should_work() {
+	let kusama_asset_metadata =
+		create_asset_metadata("Kusama", "KSM", 12, 1u128, None, false, false);
+	let kusama_location = AssetLocation(VersionedMultiLocation::V1(MultiLocation::parent()));
+
+	let manta_asset_metadata =
+		create_asset_metadata("Manta", "MANTA", 18, 1u128, None, false, false);
+	let mantan_location = AssetLocation(VersionedMultiLocation::V1(MultiLocation::new(
+		1,
+		X2(Parachain(2015), GeneralKey(b"MANTA".to_vec())),
+	)));
+	new_test_ext().execute_with(|| {
+		// Register relay chain native token
+		assert_ok!(AssetManager::register_asset(
+			Origin::root(),
+			kusama_location.clone(),
+			kusama_asset_metadata.clone()
+		));
+		assert_eq!(
+			AssetIdLocation::<Runtime>::get(0),
+			Some(kusama_location.clone())
+		);
+
+		// Register manta para chain native token
+		assert_ok!(AssetManager::register_asset(
+			Origin::root(),
+			mantan_location.clone(),
+			manta_asset_metadata.clone()
+		));
+		assert_eq!(
+			AssetIdLocation::<Runtime>::get(1),
+			Some(mantan_location.clone())
+		);
+
+		let new_location = AssetLocation(VersionedMultiLocation::V1(MultiLocation::new(
+			1,
+			X2(Parachain(1999), GeneralKey(b"UNKNOWN".to_vec())),
+		)));
+
+		// new location should be filtered
+		assert!(!crate::Pallet::<Runtime>::contains(&new_location));
```

Comment:
I'm not very familiar with the XCM code yet, but is this actually testing any "filtering"?

The way i'm reading this logic flow `new_location` is just a local variable that hasn't been given to `AssetManager::register_asset` yet (which presumably is where any filtering of foreign assets would happen), so the assert above will always be satisfied unless `new_location` happens to be a clone of `kusama_location` or `mantan_location`

I think I'm mainly confused by the "filtering" naming of the PR. Maybe you could add some description text to it to elaborate. The way I understand the code, the above test doesn't check filtering, it's only a sanity check/unit test for the `contains` implementation.

Reply:
filter means white-list. 
I agree the test should be in an end to end manner.